src/components/common/ui/form/input-mui/util/index.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  CheckCircleIcon,
+  ExclamationCircleIcon,
+  EyeIcon,
+  EyeSlashIcon,
+  MagnifyingGlassIcon,
+  XMarkIcon,
+} from '@heroicons/react/24/outline';
+
+import {
+  InputState,
+  InputType,
+} from '@/components/common/ui/form/input-mui/types';
+
+export const getState = (
+  disabled: boolean,
+  isTouched: boolean,
+  error: string,
+  isSuccessOnDefault: boolean,
+): InputState => {
+  if (disabled) return InputState.DISABLED;
+  else if (isTouched && error) return InputState.ERROR;
+  else if (isTouched && isSuccessOnDefault) return InputState.SUCCESS;
+  else return InputState.DEFAULT;
+};
+
+export const getLeftIcon = (type: InputType) =>
+  type === InputType.SEARCH ? MagnifyingGlassIcon : null;
```

Comment:
Firstly, you can use `&&` instead ternary operator. 
Secondly, it seems not like difficult logic to keep it in separate function util. Checkout comment in component